pkg/controller/dataimportcron-controller.go

```diff
@@ -336,8 +336,10 @@ func (r *DataImportCronReconciler) update(ctx context.Context, dataImportCron *c
 	}
 
 	handlePopulatedPvc := func() error {
-		if err := r.updateSource(ctx, dataImportCron, pvc); err != nil {
-			return err
+		if pvc != nil {
```

Comment:
Instead of checking here, check in updateSource if the object is nil? that way other callers are also protected. If it is nil, just return nil as you can't update the object anyway.

Reply:
We're still not sure exactly what is going on.. but once we know this fixes it we can do that

Reply:
So if we want to check for nil in updateSource (receives the `client.Object` interface),
we must use reflect, which I am not sure is worth it.. wdyt?
(reflect.ValueOf(i).IsNil())

Reply:
I personally don't see the problem with using reflect but most of the uses of `updateSource` already check the object before calling the function. I think it'd be enough to add the check here.

Reply:
I like this fix as-is